projects/alembic/alembic_dump_info_fuzzer.cc

```diff
@@ -0,0 +1,263 @@
+
+  for (size_t p = 0; p < meshPropertyCount; p++) {
+    const PropertyHeader &header = schema.getPropertyHeader(p);
+    const PropertyType pType = header.getPropertyType();
+    const std::string &name = header.getName();
+
+    if (name == "P") {
+      schema.getNumSamples();
+    } else if (name == "N") {
+      schema.getNormalsParam().getNumSamples();
+    } else if (name == "uv" || name == "st") {
+      schema.getUVsParam().getNumSamples();
+    } else if (name == ".arbGeomParams") {
+      // additional geometry elements (color sets, additional texture
+      // coordinates)
+      const ICompoundProperty geoParam = schema.getArbGeomParams();
+      const size_t geoPropCount = geoParam.getNumProperties();
+
+      for (size_t g = 0; g < geoPropCount; g++) {
+        const PropertyHeader &headerGeo = geoParam.getPropertyHeader(g);
+        const std::string &nameGeo = headerGeo.getName();
+      }
+    }
+  }
+}
+
+void dumpSubDAttributes(const ISubDSchema &schema) {
+  const size_t subdPropertyCount = schema.getNumProperties();
+
+  for (size_t p = 0; p < subdPropertyCount; p++) {
+    const PropertyHeader &header = schema.getPropertyHeader(p);
+    const PropertyType pType = header.getPropertyType();
+    const std::string &name = header.getName();
+
+    if (name == "P") {
+      schema.getNumSamples();
+    } else if (name == "uv" || name == "st") {
+      schema.getUVsParam().getNumSamples();
+    } else if (name == ".arbGeomParams") {
+      // additional geometry elements (color sets, additional texture
+      // coordinates)
+      const ICompoundProperty geoParam = schema.getArbGeomParams();
+      const size_t geoPropCount = geoParam.getNumProperties();
+
+      for (size_t g = 0; g < geoPropCount; g++) {
+        const PropertyHeader &headerGeo = geoParam.getPropertyHeader(g);
+        const std::string &nameGeo = headerGeo.getName();
+      }
+    }
+  }
+}
+
+void dumpCurvesAttributes(const ICurvesSchema &schema) {
+  const size_t propertyCount = schema.getNumProperties();
+
+  for (size_t p = 0; p < propertyCount; p++) {
+    const PropertyHeader &header = schema.getPropertyHeader(p);
+    const PropertyType pType = header.getPropertyType();
+    const std::string &name = header.getName();
+
+    if (name == "P") {
+      schema.getNumSamples();
+    } else if (name == "N") {
+      schema.getNormalsParam().getNumSamples();
+    } else if (name == "uv" || name == "st") {
+      schema.getUVsParam().getNumSamples();
+    } else if (name == ".arbGeomParams") {
+      // additional geometry elements (color sets, additional texture
+      // coordinates)
+      const ICompoundProperty geoParam = schema.getArbGeomParams();
+      const size_t geoPropCount = geoParam.getNumProperties();
+
+      for (size_t g = 0; g < geoPropCount; g++) {
+        const PropertyHeader &headerGeo = geoParam.getPropertyHeader(g);
+        const std::string &nameGeo = headerGeo.getName();
+      }
+    }
+  }
+}
```

Comment:
Maybe I'm not seeing some important dissimilarity, but it looks to me like you could factor some shared code out of `dumpMeshAttributes`, `dumpSubDAttributes`, and `dumpCurvesAttributes`.